app/uk/gov/hmrc/accessibilitystatementfrontend/parsers/YamlParser.scala

```diff
@@ -24,10 +24,13 @@ import scala.io.Source
 import scala.util.{Failure, Success, Try}
 
 class YamlParser[T: Decoder] extends Logging {
-  def parse(yaml: String): Either[Error, T] =
+  def parse(yaml: String): Either[Exception, T] =
     circeYaml.parser
       .parse(yaml)
-      .flatMap(_.as[T])
+      .fold(
```

Comment:
I believe this approach addresses the concern of our tests inadvertently verifying Circe's error messages, which might appear as though we're testing Circe itself. By handling Circe errors and converting them to our custom exception type, we're ensuring that our tests focus on our own implementation. This way, if Circe's behavior changes in the future, it will cause our tests to fail appropriately, whereas simply testing for a Left could lead to false positives.